src/shogun/util/utils.h

```diff
@@ -0,0 +1,38 @@
+/*
+* Written (W) 2019 Giovanni De Toni
+*/
+
+#ifndef SHOGUN_UTILS_H_H
+#define SHOGUN_UTILS_H_H
+
+#include <shogun/lib/common.h>
+#include <limits>
+#include <typeinfo>
+
+namespace shogun {
+
+	/**
+	 * Cast an input size_t value to the int32_t type.
+	 * Throws an exception if something goes wrong.
+	 * @code
+	 * unsigned long o = 1000000000000;
+	 * try {
+	 *	int a = size_t_to_int32_cast(o);
+	 * } catch (std::exception e)
+	 * {
+	 *	std::cout << e.what() << std::endl;
+	 * }
+	 * @endcode
+	 * @param v original value
+	 * @return casted value
+	 */
+	int32_t size_t_to_int32_cast(size_t v) {
```

Comment:
@karlnapf were you thinking about something like this?

Reply:
Force it inline

Reply:
Updated.